src/S3/ObjectCopier.php

```diff
@@ -74,8 +77,15 @@ public function __construct(
     public function promise()
     {
         return \GuzzleHttp\Promise\coroutine(function () {
+            $headObjectCommand = $this->client->getCommand(
+                'HeadObject',
+                $this->options['params'] + $this->source
+            );
+            if (is_callable($this->options['before_lookup'])) {
```

Comment:
Fari enough. Please update the changelog message to reflect that this might change the behavior of the client.